### PR TITLE
Increase descent rate at 250 knots below 10,000 feet

### PR DESCRIPTION
**Pull request by donstim**
Fixes issue of inadequate descent rate below 10,000 feet altitude.

Possible fix for #2203 

## Summary of Changes
Climb and descent performance was checked against the FCOM. Performance matched pretty well, except for descent below 10,000 feet. (Also climb above the crossover speed was off a bit, but will wait for new engine model and new autopilot to address this one.) This PR reduces idle thrust in a small corner of the envelope to result in accurate descent performance for idle thrust descents mainly below 10,000 feet (at 250 knots) without impacting other areas.

## References
Screenshot (413)
From the above data, it can be seen that it should take 3:24 and 16 miles to descend from 10,000 feet to 5,000 feet at 250 knots and 65 T weight. It should take a further 2:36 and 11 miles to descend from 5,000 feet to 1,500 feet for a total of 6 minutes and 27 miles. With this modification, I was able to match these numbers within a few seconds. Without this modification, it took around 4:35 and 21 miles to descend from 10,000 feet to 5,000 feet and around 3:23 and 15 miles from 5,000 to 1,500 feet for a total of nearly 8 minutes and 36 miles.

## Additional context
[**Revised**] How this was developed and verified to match the FCOM performance data (this is for information only - you don't necessarily need to do this for QA testing but it would be nice if at least one tester does this to confirm my numbers):
1. Set "clear skies" preset. Verify that there are no winds, and ISA day (temp 15 deg c/60 deg F at sea level and pressure 1013/29.92)
2. Set the airplane up at 12,000 feet, 250 KCAS at 64.9 T on autopilot/autothrust. The purpose of setting up at 12,000 feet is to have the airplane descending at idle thrust (N1 around 23.5%) at 250 knots at 10,000 feet.
3. Be in dev mode, open a project, and select the aircraft editor tool
4. Turn the altitude knob down as far as it will go (0100) and engage open descent (selected altitude) mode.
6. When the airplane gets to 10,000 feet, start the chrono, and if you choose to verify the distance, open the debug "tracking" window.
7. When the airplane gets to 5000 feet, record the elapsed time from the chronometer and the distance traveled from the tracking debug window.
8. When the airplane get to 1,500 feet, again record the elapsed time from the chronometer and the distance traveled from the tracking debug window.
9. Compare to FCOM performance data.
10. 10. Be aware that the autopilot does not hold speed all that well below 5000 feet, but don't try to do anything about it. It should be flying speed on pitch, but it doesn't seem to do that particularly well at times.

**For QA testing**, make sure nothing else was affected (e.g., taxi), and that you get a reasonable descent rate at 250 knots below 10,000 feet (around 1300-1500 fpm) at 64.9 T and ISA temps/no winds  compared to around 1100-1200 fpm with the current dev version . Keep in mind that this will mod will only affect thrust levels at 20-25% N1 at Mach 0.4-0.5  (220-300 knots depending on altitude and temperature for altitudes of 10,000 feet and below).

Discord username (if different from GitHub):donbikes#4084

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

**Comment by St54Kevin** [comment]
Also remember to set ISA conditions, so 15 degrees Celcius at MSL and 1013hPa/29.92

**Comment by donstim** [comment]
Using the clear skies weather preset does that for you.

**Review by wpine215 — APPROVED** [review]

**Comment by MisterChocker** [comment]
Keep in mind this has also influence on deceleration at approach

**Comment by donstim** [comment]
> Keep in mind this has also influence on deceleration at approach

Only when you are at idle thrust.

**Comment by MisterChocker** [comment]
> > Keep in mind this has also influence on deceleration at approach
> 
> Only when you are at idle thrust.

has the taxi performance changed?

**Comment by donstim** [comment]
> > > Keep in mind this has also influence on deceleration at approach
> > 
> > 
> > Only when you are at idle thrust.
> 
> has the taxi performance changed? or did you left the low Ms untouched?

Please see the write-up under "Additional context" starting with "For QA testing..."  The answer you seek is in there! xD:

**Comment by CaptainR3X** [comment]
Just started testing, backup alt indicator is broken
image

**Comment by wpine215** [comment]
> 
> 
> Just started testing, backup alt indicator is broken
> image

...not related to this PR

**Comment by ChruutvoLuzi** [comment]
Category: QA Tester
Name: ChruutvoLuzi | SD-PB.ch#8902
Date of testing: 16.12.2020
Version of the sim: 1.11.7.0
PR Tested: #2398 
Object of test: Increased descent rate below 10000ft

Testing technique: Followed the instructions of this PR
Steps you've made to test the Pull Request:
1. Spawned on a clear day, ISA, 15° C at MSL and deleted all winds
2. Slewed up to 10000ft
3. turned on AP and waited until the aircraft stabilised at 250kts IAS
4. selected open climb to 100ft and turned chrono and debug window on
5. took screenshots at 5000ft and 1500ft
6. tested taxiing at various airports

10000ft initial setup
Microsoft Flight Simulator Screenshot 2020 12 16 - 19 53 32 23

5000ft
Microsoft Flight Simulator Screenshot 2020 12 16 - 19 56 25 47

1500ft
Microsoft Flight Simulator Screenshot 2020 12 16 - 19 59 04 27


Flight notes (If required): Descent was between 1500 and 1300 fpm as described

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Tested, works as described, no issues found

**Comment by CaptainR3X** [comment]
Category: QA Tester
Name: GustavRex #4347
Date of testing: Current date 16/12/2020
Version of the sim: 1.11.7.0
PR Tested: #2398
Object of test: Test increased rate of descent at 250kts below 10,000ft

Testing technique: Basic test from FL100 and Flight; EKCH-ESGG
Steps you've made to test the Pull Request:
Basic test from FL100
1. Spawned in 65T clear skies, slewed to FL100 at 300KCAS
2. Decreased alt on dial to minimum, then and at 250kt engaged open des
3. Noting chrono time at 5000ft and 1500ft 

Flight
1. C/D EKCH, preflight programming, P/S and taxi - checking that idle thrust should taxi on its own and relatively fast
2. T/O, departure, climb and crz 
3. Arrival programming, descent and approach (cross-checking descent rate under FL100/250kt)

Flight notes (If required): Online, VatSim

Issues (If Any): None
Overall Rating: LGTM
Conclusions of the testing:
Chrono time:
5000ft: 3mins 43sec
1500ft: 6mins 16sec total -> 2min 33sec from 5000-1500
Descent was between 1300-1500 fpm as referred 

Flight test
Taxi seemed normal
Descent was better bel 10000ft

Works well!

**Comment by donstim** [comment]
As a result of the QA testing, I decided I wanted to get a little closer to the FCOM descent time of 6 minutes between 10,000 feet and 5,000 feet (compared to the 6:15-6:16 found by the testers (thanks!). Then, at the urging of @MisterChocker (which I admit I resisted quite a bit), I also did a pretty sizable set of tests to verify that deceleration rates were not significantly adversely affected by the change in idle thrust. Partly due to this testing and the realization that the Mach 0.3 idle thrust level has no effect on the descent performance at the FCOM speeds, I removed the idle thrust change at Mach 0.3. This further ensures that there is little to no effect on deceleration rates or descent rates with and flap configuration other than flaps up.

The deceleration checks consisted of the following at 2 weights -- 60 T and 65 T:
1. 300 knots to 250 knots at 10,000 ft (flaps up)
2. 250 knots to green dot speed at 10,000 ft and at 5,000 ft (Flaps up)
3. GD speed to S speed at 10,000 ft and at 5,000 ft (CONF 1)
4. S speed to F speed at 5000 ft (CONF 2) both gear up and gear down
5. S speed to F speed descending from 5000 ft at 700 fpm (CONF 2 with gear down)

As mentioned above, the deceleration time with and without this PR for all configurations other than flaps up -- conditions 3-5 -- were the same. The thrust changes in this PR had no effect.

Condition 1 is the only condition for which we have Airbus data in the FCOM -- at 65 T. The FCOM data indicates a deceleration time of 0.8 minutes - which is 48 seconds, with a precision level of 6 seconds. Without this PR, the time is 50 seconds and with it, the time is 44 seconds. Neither are significantly different than the FCOM time of 48 seconds.

The most critical condition to check for a deceleration time difference pre-PR and post-PR would be where there is a large deceleration needed at speeds between Mach 0.4 and 0.5. Of the conditions tested, the condition at 60 T, going from 250 knots (Mach 0.413) to the Green Dot  speed of 205 knots (Mach 0.339) at 5000 feet altitude had the largest difference in deceleration time. Pre-PR, the time is 1:15, post-PR it is 1:00. There could be a larger difference at lower weights due to the lower Green Dot speed, but 205 knots is already close to the Mach 0.3 point where there is no change to the idle thrust level.

At the 65 T weight, the Green Dot speed is closer to 250 knots, so the difference in decel times is about 1/2 what it is at 60 T. 

In conclusion, there is no difference in deceleration rates at operational speeds in any configuration other than flaps up, and there are only minor differences for most flaps up conditions (for which we don't have any data other than the one condition where it matches fairly well).

**Comment by ChruutvoLuzi** [comment]
Category: QA Tester
Name: ChruutvoLuzi | SD-PB.ch#8902
Date of testing: 17.12.2020
Version of the sim: 1.11.7.0
PR Tested: #2398 
Object of test: increased descent rate below 10000ft @ 250kts

Testing technique: tested in clean testing environment (ISA, GW65t, no winds)
Steps you've made to test the Pull Request:
1. Spawned and slewed up to 10000ft
2. stabilized aircraft and started open descend at 250kts, started the chrono
3. timed at 5000ft and 1500ft

Flight notes (If required):
Times: 10000ft to 5000ft: 3min 30s
          5000ft to 1500ft: 2min 26s
Total: 5min 56s
Descent rate was between 1600 initially and 1500fpm
The Aircraft slightly overshooted the speed target by:
         10000ft to 5000ft: 6kts
         5000ft to 1500ft:   4kts
Total Overshooting from 10000ft to 1500ft due to Asobo AP Pitch: 10kts (260kts)

Microsoft Flight Simulator Screenshot 2020 12 17 - 21 09 34 47
Microsoft Flight Simulator Screenshot 2020 12 17 - 21 12 00 68

Issues (If Any): None
Overall Rating: Good
Conclusions of the testing: Aligns with the datas in the FCOM, looks good

**Review by wpine215 — APPROVED** [review]

**Comment by phildeg31** [comment]
Thank you @donstim for this correction. I will check and compare tomorrow with the flight done today in real A320 simulator. 

A video of descent before landing in managed mode has been put in ticket #2203.

**Comment by donstim** [comment]
Ok @phildeg31 . I will take a look at it in a bit.
